js/packages/seed-vault/src/useSeedVault.ts

```diff
@@ -52,7 +52,7 @@ const checkIsSeedVaultAvailable = async (allowSimulated: boolean = false) => {
 
 const SEED_VAULT_EVENT_BRIDGE_NAME = 'SeedVaultEventBridge';
 
-export function subscribeToNativeEvents(
+export function useSeedVault(
     handleSeedVaultEvent: (event: SeedVaultEvent) => void,
     handleContentChange: (event: SeedVaultContentChange) => void,
```

Comment:
So, there's a way of designing this hook so that the `useEffect` doesn't constantly fire. You can bet your bottom dollar that people are going to do this:

```ts
useSeedVault(
  e => // handle seed vault event,
  e => // handle content change,
);
```

Every single time that code runs, the non-memoized functions will be _different_ which means that the `useEffect's` dependencies will be busted which means that the `remove()` will fire and the `NativeEventEmitter` will be set up all over again – on every render.

So here's the thing. Since these functions will _only_ ever get called between renders (ie. when the native event emitter fires) you can:

1. Store the functions in a `ref` – refs do _not_ have to be part of the `useEffect's` dependencies
2. Update that ref on every render
3. Reach into the ref when you need the function to call

That way, the `useEffect` will only ever run once to set up the listener, and once when the whole thing unmounts.

https://legacy.reactjs.org/docs/hooks-faq.html#how-can-i-measure-a-dom-node Keyword: ‘Keep latest props in a ref’